File: Java/No1.java
void main(){
        String a = "Salam men hemid";
        Map<Character, Integer> d = new HashMap<>();
        List<Character> keys = new ArrayList<>();
        
        for (char i : a.toCharArray()) {
            if (i != ' ') {
                d.put(i, d.getOrDefault(i, 0) + 1);
            }
        }
        

        keys.addAll(d.keySet());
        
        for (int i = 0; i < keys.size(); i++) {
            for (int j = i + 1; j < keys.size(); j++) {
                char keyI = keys.get(i);
                char keyJ = keys.get(j);
                if (d.get(keyI) > d.get(keyJ)) {
                    keys.set(i, keyJ);
                    keys.set(j, keyI);
                } else if (d.get(keyI).equals(d.get(keyJ))) {
                    if (keyI > keyJ) {
                        keys.set(i, keyJ);
                        keys.set(j, keyI);
                    }
                }
            }
        }
        
        Map<Character, Integer> new_d = new LinkedHashMap<>();
        for (char k : keys) {
            new_d.put(k, d.get(k));
        }
        
        System.out.println(new_d);
        
        StringBuilder c = new StringBuilder();
        for (Map.Entry<Character, Integer> entry : new_d.entrySet()) {
            for (int i = 0; i < entry.getValue(); i++) {
                c.append(entry.getKey());
            }
        }
        
        IO.print(c.toString());
    }

File: Python/No1.py
a = "Salam men hemid"
d = {}
keys=[]
for i in a:
    if i != ' ':
        if i in d:
            d[i] += 1
        else:
            d[i] = 1


for i in d:
    keys+=[i]


for i in range(len(keys)):
    for j in range(i + 1, len(keys)):
        if d[keys[i]] > d[keys[j]]:
            keys[i], keys[j] = keys[j], keys[i]
        elif d[keys[i]] == d[keys[j]]:
            if keys[i]>keys[j]:
                keys[i], keys[j] = keys[j], keys[i]

new_d = {}


for k in keys:
    new_d[k] = d[k]

print(new_d)    

c=''
for i in new_d:
    c+=i*new_d[i]

print(c)